nixos/modules/services/networking/nix-serve.nix

```diff
@@ -66,5 +66,7 @@ in
       description = "Nix-serve user";
       uid = config.ids.uids.nix-serve;
     };
+
+    nix.allowedUsers = [ "nix-serve" ];
```

Comment:
Would `nix.allowedUsers = mkOptionDefault [ "nix-serve" ]` work instead? I just tried it and I can still manipulate the nix store, so it does seem to work. I assume this will result in the default becoming [ "*", "nix-serve" ] if nix-serve is enabled, which should be harmless.